src/main/java/com/internet/shop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,158 @@
+                    + "WHERE o.user_id = ? AND o.deleted = FALSE",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            statement.setLong(1, userId);
+            return getOrdersListFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Getting orders for user with id "
+                                              + userId + " failed", e);
+        }
+    }
+
+    @Override
+    public List<Order> getAll() {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection.prepareStatement(
+                    "SELECT * FROM orders o "
+                    + "JOIN orders_products op ON o.id = op.order_id "
+                    + "JOIN products p on op.product_id = p.id "
+                    + "WHERE o.deleted = FALSE ORDER BY o.id",
+                    ResultSet.TYPE_SCROLL_INSENSITIVE, ResultSet.CONCUR_UPDATABLE);
+            return getOrdersListFromStatement(statement);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can not get items from DB", e);
+        }
+    }
+
+    @Override
+    public Order update(Order order) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteProductsFromDB(order.getId(), connection);
+            addProductsToDB(order, connection);
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE orders SET user_id = ? "
+                    + "WHERE id = ? AND deleted = FALSE");
+            statement.setLong(1, order.getUserId());
+            statement.setLong(2, order.getId());
+            statement.executeUpdate();
+            return order;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Updating order " + order + " failed", e);
+        }
+    }
+
+    @Override
+    public boolean deleteById(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            deleteProductsFromDB(id, connection);
+            PreparedStatement statement = connection.prepareStatement(
+                    "UPDATE orders SET deleted = TRUE WHERE id = ?");
+            statement.setLong(1, id);
+            return statement.executeUpdate() == 1;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Deleting order with id " + id + " failed", e);
+        }
+    }
+
+    private Order getOrderFromResultSet(ResultSet resultSet) throws SQLException {
+        long id = resultSet.getLong("id");
+        long userId = resultSet.getLong("user_id");
+        List<Product> products = new ArrayList<>();
+        do {
+            products.add(new Product(resultSet.getLong("product_id"),
+                    resultSet.getString("name"), resultSet.getDouble("price")));
+        } while (resultSet.next() && id == resultSet.getLong("id"));
```

Comment:
what is the purpose of do while here? is it possible that we have an empty result set?

Reply:
Can't understand your comment. Of course I don't expect empty set. While is necessary to call next() multiple times.